WordPress/gutenberg
# [FSE - Templates]: Add info to templates created from wp suggestion

## What?
Part of: https://github.com/WordPress/gutenberg/issues/37407
Needed for: https://github.com/WordPress/gutenberg/pull/41189

This PR adds a post meta(`is_wp_suggestion`) to templates created from the site editor.


## Why?
Currently we handle some template actions/filtering based on whether the template is part of the template hierarchy. This currently happens through `get_default_block_template_types` results, that are hardcoded and of course are just a few of the possible templates. Many specific templates will need to be dynamically calculated and suggested(see linked PR for single-$post_type, archive-$post_type and post specific templates) and these should be treated as well as part of the template hierarchy.

In a few words we need to differentiate the templates created from the post editor in the `Template` panel in inspector controls and the ones suggested in site editor. So in the suggested templates in site editor list(`Add new`) from WP we should mark them as suggested.

## Testing instructions
The two main differences are:
1. in site editor we should not be able to rename a template from template hierarchy
2. such a template should not be shown in post editor available templates
3. Create a new template from post editor
4. Add the below snippet [here(before the return)](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/add-new-template/new-template.js#L154) and create the template
`missingTemplates.push( {
	title: 'Suggested',
	description: 'test template',
	slug: 'single-book',
} );

3. Test that everything is working as expected and no regression is introduced
6. Existing templates should rename unaffected.
